diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SortRule.kt

```diff
@@ -0,0 +1,112 @@
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.isClassEnum() && node.hasChildOfType(CLASS_BODY))
+            sortEnum(node.findChildByType(CLASS_BODY)!!)
+        if (node.isCompanionObject() && node.hasChildOfType(CLASS_BODY))
+            sortProperty(node.findChildByType(CLASS_BODY)!!)
+    }
+
+    private fun sortProperty(node: ASTNode) {
+        val propertyList = node.getChildren(null)
+                .filter { it.elementType == PROPERTY }
+                .filter { it.hasChildOfType(MODIFIER_LIST) }
+                .filter { it.findChildByType(MODIFIER_LIST)!!.hasChildOfType(CONST_KEYWORD) }
+        if (propertyList.size <= 1)
+            return
+        val consecutivePropertiesGroups = createOrderListOfList(propertyList)
+        val sortedListOfList = mutableListOf<List<ASTNode>>()
+        consecutivePropertiesGroups.forEach { nodesList ->
+            val sortList = nodesList.sortedBy { it.findChildByType(IDENTIFIER)!!.text }
+            sortedListOfList.add(sortList)
+        }
+        consecutivePropertiesGroups.forEachIndexed { index, mutableList ->
+            if (mutableList != sortedListOfList[index]) {
+                WRONG_DECLARATIONS_ORDER.warnAndFix(configRules, emitWarn, isFixMode,
+                        "constant properties inside companion object order is incorrect",
+                        mutableList.first().startOffset) {
+                    swapSortNodes(sortedListOfList[index], mutableList, node)
+                }
+            }
+        }
+    }
+
+    private fun swapSortNodes(sortList: List<ASTNode>, nonSortList: List<ASTNode>, node: ASTNode) {
+        val nodeBefore: ASTNode? = nonSortList.last().treeNext
+        node.removeRange(nonSortList.first(), nonSortList.last().treeNext)
+        sortList.forEachIndexed { nodeIndex, astNode ->
+            if (nodeIndex != 0)
+                node.addChild(PsiWhiteSpaceImpl("\n"), nodeBefore)
+            node.addChild(astNode, nodeBefore)
+        }
+    }
+
+    private fun createOrderListOfList(propertyList: List<ASTNode>): MutableList<MutableList<ASTNode>> {
+        val orderListOfList = mutableListOf<MutableList<ASTNode>>()
+        var oneOrderList = mutableListOf(propertyList.first())
+        val propertyZipList = propertyList.zipWithNext()
+        propertyZipList.forEach { nodePair ->
+            if (nodePair.first.nextSibling { it.elementType == PROPERTY } == nodePair.second)
+                oneOrderList.add(nodePair.second)
+            else {
+                orderListOfList.add(oneOrderList)
+                oneOrderList = mutableListOf(nodePair.second)
+            }
+        }
+        orderListOfList.add(oneOrderList)
+        return orderListOfList
```

Comment:
```suggestion
        return orderListOfList.toList()
```
I think you don't need mutability outside of this method?